File: Java/labs/lab21_Strategy/task_2_1/src/Navigator.java
public class Navigator {
    private RouteStrategy strategy;

    public void setStrategy(RouteStrategy strategy) {
        this.strategy = strategy;
    }

    public void buildRoute(String A, String B){
        if (strategy == null){
            System.out.println("Стратегія не вибрана");
            return;
        }

        String route = strategy.buildRoute(A, B);
        System.out.println("Маршрут побудовано: " + route);
    }
}

File: Cpp/lab21_Strategy/task_2_1/src/navigator.cpp
#include <iostream>

#include "navigator.hpp"


void Navigator::setStrategy(RouteStrategy* strategy)
{
  this->strategy = strategy;
}

void Navigator::buildRoute(std::string A, std::string B)
{
  if (strategy == nullptr)
  {
    std::cerr << "Стратегія не вибрана" << std::endl;
  }

  std::string route = this->strategy->buildRoute(A, B);
  std::cout << "Маршрут побудовано: " << route << std::endl;
}
